src/merkle/simple_smt/tests.rs

```diff
@@ -138,6 +138,51 @@ fn get_path() {
     assert_eq!(vec![node2], *tree.get_path(NodeIndex::new(1, 1)).unwrap());
 }
 
+#[test]
+fn test_parent_node_iterator() -> Result<(), MerkleError> {
+    let tree = SimpleSmt::new(2)
+        .unwrap()
+        .with_leaves(KEYS4.into_iter().zip(VALUES4.into_iter()))
+        .unwrap();
+
+    // check depth 2
+    assert_eq!(VALUES4[0], tree.get_node(&NodeIndex::new(2, 0)).unwrap());
```

Comment:
another question, why is `SimpleSmt::get` receiving a shared `NodeIndex`? It seems the other structures receive an exclusive index. [`MerkleStore`](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/store/mod.rs#L169), [`MerkleTree`](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/merkle_tree.rs#L75)